File: CS/MapBubbleItem/Form1.cs
using System;
using System.Drawing;
using System.Windows.Forms;
using DevExpress.XtraMap;

namespace MapBubbleItem {
    public partial class Form1 : Form {
        public Form1() {
            InitializeComponent();
        }

        private void Form1_Load(object sender, EventArgs e) {
            // Create a layer to show vector items.
            VectorItemsLayer itemsLayer = new VectorItemsLayer();
            mapControl1.Layers.Add(itemsLayer);

            // Create a storage to provide data for the vector layer.
            MapItemStorage storage = new MapItemStorage();
            itemsLayer.Data = storage;

            // Create an array of Bubble items with different values, sizes and locations.
            MapBubble bubble1 = new MapBubble() { Argument = "A", Value = 200, Location = new GeoPoint(-45, -60), Size = 20 };
            MapBubble bubble2 = new MapBubble() { Argument = "B", Value = 400, Location = new GeoPoint(-45, 0), Size = 40 };
            MapBubble bubble3 = new MapBubble() { Argument = "C", Value = 800, Location = new GeoPoint(-45, 60), Size = 80 };
            storage.Items.AddRange(new MapItem[] { bubble1, bubble2, bubble3 });

            // Provide color indexes to bubbles as attributes.
            int i = 0;
            foreach (MapItem bubble in storage.Items) {
                bubble.Attributes.Add(new MapItemAttribute() { Name = "Color", Value = i });
                i++;
            }

            // Create a colorizer to provide colors for bubble items.
            ColorIndexColorizer colorizer = new ColorIndexColorizer();
            itemsLayer.Colorizer = colorizer;

            // Add colors to the colorizer.
            colorizer.ColorItems.Add(new ColorizerColorTextItem() { Color = Color.Coral, Text = "Category A" });
            colorizer.ColorItems.Add(new ColorizerColorTextItem() { Color = Color.Orange, Text = "Category B" });
            colorizer.ColorItems.Add(new ColorizerColorTextItem() { Color = Color.LightBlue, Text = "Category C" });

            // Load color indexes from bubbles via the 'Color' attribute
            colorizer.ColorIndexProvider = new ShapeAttributeColorIndexProvider() { AttributeName = "Color" };

            
            // Show a color legend.
            mapControl1.Legends.Add(new ColorListLegend() { Layer = itemsLayer });
        }

    }

}

File: VB/MapBubbleItem/Form1.vb
Imports System
Imports System.Drawing
Imports System.Windows.Forms
Imports DevExpress.XtraMap

Namespace MapBubbleItem
    Partial Public Class Form1
        Inherits Form

        Public Sub New()
            InitializeComponent()
        End Sub

        Private Sub Form1_Load(ByVal sender As Object, ByVal e As EventArgs) Handles MyBase.Load
            ' Create a layer to show vector items.
            Dim itemsLayer As New VectorItemsLayer() With {.Data = CreateData(), .Colorizer = CreateColorizer()}
            mapControl1.Layers.Add(itemsLayer)

            ' Show a color legend.
            mapControl1.Legends.Add(New ColorListLegend() With {.Layer = itemsLayer})
        End Sub

        #Region "#CreateBubbles"
        ' Create a storage to provide data for the vector layer.
        Private Function CreateData() As IMapDataAdapter
            Dim storage As New MapItemStorage()

            ' Add Bubble charts with different values, sizes and 
            ' locations to the storage's Items collection.
            storage.Items.Add(New MapBubble() With {.Argument = "A", .Value = 200, .Location = New GeoPoint(-45, -60), .Size = 20, .Group = 1, .MarkerType = MarkerType.Diamond})
            storage.Items.Add(New MapBubble() With {.Argument = "B", .Value = 400, .Location = New GeoPoint(-45, 0), .Size = 40, .Group = 2, .MarkerType = MarkerType.Plus})
            storage.Items.Add(New MapBubble() With {.Argument = "C", .Value = 800, .Location = New GeoPoint(-45, 60), .Size = 80, .Group = 1, .MarkerType = MarkerType.Cross})

            Return storage
        End Function
        #End Region ' #CreateBubbles

        ' Create a colorizer to provide colors for bubble items.    
        Private Function CreateColorizer() As MapColorizer
            Dim colorizer As New KeyColorColorizer()

            ' Add colors to the colorizer.
            colorizer.Colors.Add(Color.Coral)
            colorizer.Colors.Add(Color.Orange)
            colorizer.Colors.Add(Color.LightBlue)

            colorizer.Keys.Add(New ColorizerKeyItem() With {.Key = "A", .Name = "Category A"})
            colorizer.Keys.Add(New ColorizerKeyItem() With {.Key = "B", .Name = "Category B"})
            colorizer.Keys.Add(New ColorizerKeyItem() With {.Key = "C", .Name = "Category C"})

            ' Load color indexes from bubbles via the 'Color' attribute
            colorizer.ItemKeyProvider = New ArgumentItemKeyProvider()

            Return colorizer
        End Function
    End Class

End Namespace
